grace/server.go

```diff
@@ -240,6 +248,7 @@ func (srv *Server) serverTimeout(d time.Duration) {
 			break
 		}
 		srv.wg.Done()
```

Comment:
I may not catch your point, the defer is used for handling a panic, not decrease a waitGroup